Community-Supported/native-s3/join-parquet-and-csv-on-s3.py

```diff
@@ -8,6 +8,8 @@
 # CSV file which contains the orders that were returned by the customers
 RETURNS_DATASET = escape_string_literal("s3://hyper-dev-us-west-2-bucket/tc22-demo/returns.csv")
 
+EMPTY_S3_CREDENTIALS = "ACCESS_KEY_ID => '', SECRET_ACCESS_KEY => ''"
```

Comment:
> we now require the user to explicitly specify credentials in a s3-location constructor - even for public resources that don't require authentication.

I don't think this makes sense from a user-perspective. I think empty credentials/anonymous access should stay the default

Reply:
Agree with @vogelsgesang, furthermore, it is possible to push default access_key and secret_key to the hyper engine when starting it. 
If the Access_key and secret_key become mandatory in the query :
- The default AK and SK defined at the instance level become useless
- There is also a problem of security in the hyper log file : AK and SK should be hidden every time an external query is called)

Reply:
It is possible to provide default access keys and default secret key - this will not change. What will change is that they can be empty (which they currently are by default) and thereby implicitly mean "No Authentication". When set at startup, you will still be able to use S3_LOCATION without specifying credentials again - however if you want to access an object on S3 anonymously (i.e. without authentication), you would have to explicitly specify them as blank values with the S3_LOCATION syntax.

Depending on the Hyper version you're running, access key and secret key are obfuscated in the Hyper log when specified in a SQL query.

Reply:
Ok, got it 👍